waffle_utils/video/tools.py

```diff
@@ -0,0 +1,130 @@
+        output_dir (Union[str, Path]): Path to the output directory where the frame images will be saved.
+        frame_rate (int, optional): Frame rate of the output images. Defaults to {DEFAULT_FRAME_RATE}.
+        verbose (bool, optional): Whether to print verbose output. Defaults to False.
+    """
+
+    input_path = Path(input_path)
+    output_dir = Path(output_dir)
+
+    # Create output directory if it doesn't exist
+    if not output_dir.exists():
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    # Extract frames from the video file
+    video_capture = create_video_capture(input_path)
+    fps = video_capture.get(cv2.CAP_PROP_FPS)
+    success, image = video_capture.read()
+    count = 0
+    frame_interval = int(round(fps / frame_rate))
+    while success:
+        count += 1
+        if count % frame_interval == 0:
+            output_path = output_dir / f"{count}.jpg"
+            save_image(output_path, image)
+
+            if verbose:
+                print(f"Extracted frame {count} to {output_path}.")
+
+        success, image = video_capture.read()
+
+    video_capture.release()
+
+
+def create_video(
+    input_dir: Union[str, Path],
+    output_path: Union[str, Path],
+    frame_rate: int = DEFAULT_FRAME_RATE,
+    verbose: bool = False,
+) -> None:
+    f"""Create a Video File from a Directory of Frame Images.
+
+    Args:
+        input_dir (Union[str, Path]): Path to the input directory containing the frame images.
+        output_path (Union[str, Path]): Path to the output video file.
+        frame_rate (int, optional): Frame rate of the output video. Defaults to {DEFAULT_FRAME_RATE}.
+        verbose (bool, optional): Whether to print verbose output. Defaults to False.
+    """
+
+    input_dir = Path(input_dir)
+    output_path = Path(output_path)
+
+    # Create output directory if it doesn't exist
+    if not output_path.parent.exists():
+        make_directory(input_dir)
+
+    # Get a sorted list of frame image files
+    frames = sorted(input_dir.glob("*.jpg"))
+
+    # Load the first frame to get dimensions
+    first_frame = load_image(frames[0])
+    height, width, _ = first_frame.shape
+
+    # Initialize video writer with the desired codec, frame rate, and frame size
+    ext = output_path.suffix
+
+    if ext == ".mp4":
+        fourcc = cv2.VideoWriter_fourcc(*"mp4v")
+    elif ext == ".avi":
+        if cv2.VideoWriter_fourcc(*"MJPG") == -1:
+            fourcc = cv2.VideoWriter_fourcc(*"XVID")
+        else:
+            fourcc = cv2.VideoWriter_fourcc(*"MJPG")
+    elif ext == ".wmv":
+        fourcc = cv2.VideoWriter_fourcc(*"WMV2")
+    elif ext == ".mov":
+        fourcc = cv2.VideoWriter_fourcc(*"XVID")
+    elif ext == ".flv":
+        fourcc = cv2.VideoWriter_fourcc(*"FLV1")
+    elif ext == ".mkv":
+        fourcc = cv2.VideoWriter_fourcc(*"VP80")
+    elif ext == ".mpeg" or ext == ".mpg":
+        fourcc = cv2.VideoWriter_fourcc(*"XVID")
```

Comment:
I think using `ext -> codec mapper` is better choice.

Reply:
@ZeroAct I newly designed it to get the video extension automatically from the output video file path that users typed as the input for `create_video()`.

```python
# It doesn't have the input argument for output video extension
create_video(input_dir, output_path, frame_rate, verbose)
```

```python
# waffle_utils/video/tools.py

def create_video(input_dir, output_path, frame_rate, verbose) 

    # ...

    # Check if all files have the same extension
    files = list(input_dir.glob("*"))
    file_extensions = set()
    for file in files:
        file_extensions.add(file.suffix)
    if len(file_extensions) != 1:
        raise ValueError(f"The files in {input_dir} do not have a consistent extension.")

    # Check if the extension of the files is supported
    unique_extension = file_extensions.pop()[1:]  # Get the unique extension existing.

    # ...
```

Reply:
Done